mesonbuild/options.py

```diff
@@ -1307,7 +1307,7 @@ def initialize_from_top_level_project_call(self,
             else:
                 proj_key = key.as_root()
                 if proj_key in self.options:
-                    self.options[proj_key].set_value(valstr)
+                    self.set_option(proj_key, valstr, True)
```

Comment:
nitpick: using kw `first_invocation=True` would document what that boolean value is.

Reply:
True but I'm reusing the syntax from line 1308.